Lib/typing.py

```diff
@@ -606,9 +606,12 @@ def __init__(self, name, *constraints, bound=None,
             self.__bound__ = _type_check(bound, "Bound must be a type.")
         else:
             self.__bound__ = None
-        def_mod = sys._getframe(1).f_globals['__name__']  # for pickling
-        if def_mod != 'typing':
-            self.__module__ = def_mod
+        try:
+            def_mod = sys._getframe(1).f_globals.get('__name__', '__main__')  # for pickling
```

Comment:
I don't think it's safe to assume `__name__` is `__main__` when it's not defined here. I think other classes like enums make the class unpickable when `__name__` cannot be determined.

Reply:
I think `'__main__'` is fine. By the way a similar code is used by `TypedDict` and `NamedTuple`, so you can copy it here. Or maybe it would even make sense to factor this out in an internal helper function doing this:
```python
    try:
        module = sys._getframe(1).f_globals.get('__name__', '__main__')
    except (AttributeError, ValueError):
        module = None
```